src/main/java/ui/MenuControl.java

```diff
@@ -173,21 +192,27 @@ private void onBoardSaveAs() {
      * Prompts a BoardNameDialog and saves the current board based on its response
      */
     public final void saveBoardAs() {
-        List<PanelInfo> panelList = panels.getCurrentPanelInfos();
-        BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
-        Optional<String> response = dlg.showAndWait();
-        ui.showMainStage();
-        panels.selectFirstPanel();
+        Optional<String> response = promptForNewBoardName();
 
-        if (response.isPresent()) {
-            String boardName = response.get().trim();
-            prefs.addBoard(boardName, panelList);
-            prefs.setLastOpenBoard(boardName);
-            prefs.setLastOpenBoardPanelInfos(panelList);
-            ui.triggerEvent(new BoardSavedEvent());
-            logger.info("New board " + boardName + " saved");
-            ui.updateTitle();
-        }
+        response.ifPresent(boardName -> {
+            createAndOpenNewBoard(boardName, panels.getCurrentPanelInfos());
+        });
+    }
+
+    /**
+     * Creates a new board with the given board name and panel infos, then open that new board
+     * @param boardName
+     * @param panelList
+     */
+    public final void createAndOpenNewBoard(String boardName, List<PanelInfo> panelList) {
```

Comment:
This method should be separated to:
- `saveBoard()` - for all the `prefs` and `triggerEvent`
- `showBoard()` - for `openBoard`. `ui.updateTitle()` should actually be called in `openBoard` I believe.